c/shared/source/uforead/uforead.c

```diff
@@ -1815,6 +1834,44 @@ static int parseXMLGlifFile(ufoCtx h, xmlNodePtr cur, int tag, unsigned long *un
     return ufoSuccess;
 }
 
+static bool validCIDKeyedFont(ufoCtx h, xmlNodePtr cur){
+    /* if required key found, set as true*/
+    while (cur != NULL) {
+        char* keyName = parseXMLKeyName(h, cur);
+        if (strEqual(keyName, "com.adobe.type.CIDFontName"))
+            h->requiredCIDKeys.CIDFontName = true;
+        else if (strEqual(keyName, "com.adobe.type.Registry"))
+            h->requiredCIDKeys.Registry = true;
+        else if (strEqual(keyName, "com.adobe.type.Ordering"))
+            h->requiredCIDKeys.Ordering = true;
+        else if (strEqual(keyName, "com.adobe.type.Supplement"))
+            h->requiredCIDKeys.Supplement = true;
+        else if (strEqual(keyName, "com.adobe.type.postscriptFDArray"))
+            h->requiredCIDKeys.postscriptFDArray = true;
+        else if (strEqual(keyName, "postscriptFDArray"))
+            h->requiredCIDKeys.postscriptFDArray = true;
+        else if (strEqual(keyName, "com.adobe.type.postscriptCIDMap"))
+            h->requiredCIDKeys.postscriptCIDMap = true;
+        cur = cur->next;
+    }
+
+    /* now check all required keys */
+    if (h->requiredCIDKeys.CIDFontName == false)
+        return false;
+    else if (h->requiredCIDKeys.Registry == false)
+        return false;
+    else if (h->requiredCIDKeys.Ordering == false)
+        return false;
+    else if (h->requiredCIDKeys.Supplement == false)
+        return false;
+    else if (h->requiredCIDKeys.postscriptFDArray == false)
+        return false;
+    else if (h->requiredCIDKeys.postscriptCIDMap == false)
+        return false;
+    else
+        return true;
+}
```

Comment:
@skef I'm curious to hear what you think about this part of my code in particular whenever you have time — I thought about it and this was the best/only way I could think of checking for required keys within C. If you have a better/more efficient idea for implementation, please let me know! 😄

Reply:
I don't think this is a big deal either way, but if you wanted to reduce the second part you could turn requiredCIDKeys into a uint16_t (or whatever), make a bit flags enum for each of the keys, and set those flags with `|=` in the top part. Then at the end you just need to verify that all the bits are set, probably by having an extra enum value representing the max, and just return `requiredCIDKeys == (maxBit - 1)`.

Reply:
@skef thanks for the recommendation! I liked it much more than the way I had it. My latest commit implements it the way you described.